### PR TITLE
Improve SideNav Accessibility

### PR DESCRIPTION
**Pull request by DesignerDave**
## ✍️ Proposed changes

This increases the contrast of labels in the SideNav components, as well as improves handling of Aria roles.

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] ~I have added tests that prove my fix is effective or that my feature works~
- [x] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

- [ ] ~I have added my new package to the global tsconfig~
- [ ] ~I have added my new package to the Table of Contents on the global README~

### CODE REVIEW

File: packages/side-nav/README.md [hunk 1]
```diff
@@ -47,7 +47,7 @@ return (
     <li>
       <h4 class="leafygreen-ui-19w7vet">States</h4>
       <ul role="menu" class="leafygreen-ui-bdnco">
-        <li role="menuitem">
+        <li role="none">
           <button
```
Comment (brookescarlett) [inline packages/side-nav/README.md]: Shouldn't the buttons then have role="menuitem" ? 
See HTML source doe here: https://www.w3.org/TR/wai-aria-practices-1.1/examples/menu-button/menu-button-links.html

**Comment by brookescarlett** [comment]
The implementation in the .tsx file looks correct to me, but the menuitem roles are missing in the markdown

File: packages/side-nav/README.md (hunk 1 above)
Comment (seanmhanson) [reply packages/side-nav/README.md]: Oops, @DesignerDave good catch. From W3's spec, an example:
```
<li role="none">
  <a role="menuitem" href="https://www.w3.org/standards/webdesign/accessibility">
    W3C Web Accessibility Initiative
  </a>
</li>
```
So by default if `<ul role="menu">` is your container, `li` elements treated as menu items since they're direct descendants. Since the actual interactive element is the link/button/whatever, we have to reset the role back to `role="none"`.

Does that sound right?

File: packages/side-nav/src/SideNav.spec.tsx [hunk 2]
```diff
@@ -63,10 +63,7 @@ describe('packages/side-nav', () => {
       });
 
       test('it sets the correct role for the item', () => {
-        expect(renderedEls.itemEl?.parentNode).toHaveAttribute(
-          'role',
-          'menuitem',
-        );
+        expect(renderedEls.itemEl).toHaveAttribute('role', 'menuitem');
```
Comment (seanmhanson) [inline packages/side-nav/src/SideNav.spec.tsx]: I would prefer this block:
```
test('it sets the correct role for the item and its container', () => {
   expect(renderedEls.itemEl).toHaveAttribute('role', 'menuitem');
   expect(renderedEls.itemEl?.parentNode).toHaveAttribute('role', 'none');
});
```

Just so the test coverage maintains this in the future, otherwise I could see myself or someone else modifying this without thinking once we've all stopped thinking so much about IA Navigation.

File: packages/side-nav/src/SideNavGroup.tsx [hunk 3]
```diff
@@ -5,9 +5,10 @@ import { uiColors } from '@leafygreen-ui/palette';
 import { ulStyleOverrides, SIDE_OFFSET } from './styles';
 
 const sideNavLabelStyle = css`
-  color: ${uiColors.green.base};
+  font-size: 11px;
```
Comment (seanmhanson) [inline packages/side-nav/src/SideNavGroup.tsx]: So this is more push-back against the design than this PR, but looking at this in context right now on Cloud, `11px` is borderline legible. I really think `12px` is the smallest we should go for something that will show up on nearly every page of every application.

Example: I have a moderate prescription for glasses (nearsighted + astigmatism), and without my glasses the letters of the side nav group header at 11px start to blur into each other, about one foot from the monitor.

Take it or leave it, just a thought.

File: packages/side-nav/src/SideNavGroup.tsx [hunk 4]
```diff
@@ -5,9 +5,10 @@ import { uiColors } from '@leafygreen-ui/palette';
...
+  font-weight: bold;
```
Comment (seanmhanson) [inline packages/side-nav/src/SideNavGroup.tsx]: Thank you for the drive-by on behalf of legacy less code

File: packages/side-nav/README.md (hunk 1 above)
Comment (DesignerDave) [reply packages/side-nav/README.md]: Ahh, yep! Sorry, I believe this is implemented correctly, just forgot to add the role in the README.

File: packages/side-nav/src/SideNav.spec.tsx (hunk 2 above)
Comment (DesignerDave) [reply packages/side-nav/src/SideNav.spec.tsx]: Great, will do.

File: packages/side-nav/src/SideNavGroup.tsx (hunk 4 above)
Comment (DesignerDave) [reply packages/side-nav/src/SideNavGroup.tsx]: 👍

File: packages/side-nav/README.md (hunk 1 above)
Comment (brookescarlett) [reply packages/side-nav/README.md]: Shouldn't the button have role="menuitem"?

Comment (brookescarlett) [reply packages/side-nav/README.md]: I only see the role set on the LI

**Comment by brookescarlett** [comment]
@DesignerDave just noticed that the readme doesn't have the new Installation instructions we added everywhere else, if you don't mind adding while you're in here

File: packages/side-nav/src/SideNavGroup.tsx (hunk 3 above)
Comment (DesignerDave) [reply packages/side-nav/src/SideNavGroup.tsx]: Rob is working on typography guidelines which will give us the justification to change this. As of now, we should probably stick with the mocks on size, though I did work with Nelle to increase the contrast.

We would need to make some larger modifications to the spacing around the label to maintain the visual hierarchy to make the change to a larger font-size, but I do agree with you. I think we should go forward with this.

That said, if we add `letter-spacing: 0.3px;` to the component, does this address your concerns at all?

Comment (seanmhanson) [reply packages/side-nav/src/SideNavGroup.tsx]: Sure thing. If this is being addressed overall with guidelines, I'm good with whatever we need for now.

File: packages/side-nav/README.md (hunk 1 above)
Comment (DesignerDave) [reply packages/side-nav/README.md]: I'm seeing it as an attribute of the button in the PR. It was already there prior, so it's not showing up as a change.

Comment (brookescarlett) [reply packages/side-nav/README.md]: got it.. the compare view was cutting a bit out so I didn't notice
